### ISSUE TITLE
Remove the Debug jobs from the official build, only running them in PR validation

### ISSUE THREAD

**Issue by dagood** [dotnet/core-setup#8404]
The Debug jobs in the official build take resources, and we should remove them if they don't serve a practical purpose to reduce waste and contention.

The only situation I can think of where they are useful is if two PRs merge at around the same time and cause a test failure *only in the Debug configuration*. For example, a dependency update and host change coming in at the same time: there is no Git conflict, but the PR changes weren't tested against each other and could break together. I don't know if that's a practical concern.

@vitek-karas, @elinor-fung, do you know if there are other reasons to keep the Debug jobs in the official builds active? Is the above scenario a concern?

/cc @mmitche

**Comment by vitek-karas** [comment]
I had no idea we even do that - core-setup doesn't have outerloop, does it? So all tests run during CI. In which case I'm perfectly fine having Debug runs only during CIs.

**Comment by dagood** [comment]
That's right, the tests all run during every CI and PR validation build, so barring the weird scenario above, there's no loss of coverage.

I don't have time to follow through on this at the moment, but @mmitche if you're interested I believe it'd be just some template `if`s on the build yaml matrix values.

**Comment by jkoritzinsky** [comment]
The public CI builds for core-setup are currently disabled on AzDO, so the official build coverage is currently the only CI test coverage (and that’s a batched/rolling build).
